File: android/app/src/main/kotlin/com/alex/cabogame/models/Card.kt
package com.alex.cabogame.models

import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
enum class Suit { HEARTS, DIAMONDS, CLUBS, SPADES }

@Serializable
enum class Rank(val value: Int) {
    ACE(1), TWO(2), THREE(3), FOUR(4), FIVE(5), SIX(6), SEVEN(7),
    EIGHT(8), NINE(9), TEN(10), JACK(11), QUEEN(12), KING(13);

    val display: String get() = when (this) {
        ACE -> "A"; JACK -> "J"; QUEEN -> "Q"; KING -> "K"
        else -> value.toString()
    }

    val caboValue: Int get() = value
}

@Serializable
data class Card(
    val id: String = UUID.randomUUID().toString(),
    val suit: Suit,
    val rank: Rank
) {
    val shortName: String get() {
        val sym = when (suit) {
            Suit.HEARTS -> "♥"
            Suit.DIAMONDS -> "♦"
            Suit.CLUBS -> "♣"
            Suit.SPADES -> "♠"
        }
        return "${rank.display}$sym"
    }
}

fun caboDeck(shuffled: Boolean = true): List<Card> {
    val deck = Suit.entries.flatMap { suit -> Rank.entries.map { rank -> Card(suit = suit, rank = rank) } }
    return if (shuffled) deck.shuffled() else deck
}

File: ios/CaboGame/Models/Card.swift
import Foundation

enum Suit: String, CaseIterable, Codable {
    case hearts
    case diamonds
    case clubs
    case spades
}

enum Rank: Int, CaseIterable, Codable {
    case ace = 1
    case two = 2
    case three = 3
    case four = 4
    case five = 5
    case six = 6
    case seven = 7
    case eight = 8
    case nine = 9
    case ten = 10
    case jack = 11
    case queen = 12
    case king = 13

    var display: String {
        switch self {
        case .ace: return "A"
        case .jack: return "J"
        case .queen: return "Q"
        case .king: return "K"
        default: return "\(rawValue)"
        }
    }

    var caboValue: Int {
        rawValue
    }
}

struct Card: Codable, Equatable, Identifiable {
    let id: UUID
    let suit: Suit
    let rank: Rank

    init(id: UUID = UUID(), suit: Suit, rank: Rank) {
        self.id = id
        self.suit = suit
        self.rank = rank
    }

    var shortName: String {
        let suitSymbol: String
        switch suit {
        case .hearts: suitSymbol = "♥"
        case .diamonds: suitSymbol = "♦"
        case .clubs: suitSymbol = "♣"
        case .spades: suitSymbol = "♠"
        }
        return "\(rank.display)\(suitSymbol)"
    }
}

extension Array where Element == Card {
    static func caboDeck(shuffled: Bool = true) -> [Card] {
        var deck: [Card] = []
        for suit in Suit.allCases {
            for rank in Rank.allCases {
                deck.append(Card(suit: suit, rank: rank))
            }
        }
        return shuffled ? deck.shuffled() : deck
    }
}
